components/centraldashboard-angular/frontend/src/app/services/backend.service.ts

```diff
@@ -0,0 +1,98 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { envInfo } from '../types/env-info';
+import { SnackBarService } from './snack-bar/snack-bar.service';
+import { SnackType } from './snack-bar/types';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CDBBackendService {
+  constructor(private http: HttpClient, private snackBar: SnackBarService) {}
+
+  public getEnvInfo(): Observable<envInfo> {
+    const url = `api/workgroup/env-info`;
+
+    return this.http
+      .get<envInfo>(url)
+      .pipe(catchError(error => this.handleError(error)));
+  }
+
+  // ---------------------------Error Handling---------------------------------
+  public handleError(
+    error: HttpErrorResponse | ErrorEvent | string,
+    showSnackBar = true,
+  ): Observable<never> {
+    // The backend returned an unsuccessful response code.
+    // The response body may contain clues as to what went wrong,
+    console.error(error);
+    if (showSnackBar) {
+      this.snackBar.open(this.getSnackErrorMessage(error), SnackType.Error);
+    }
+
+    return throwError(this.getErrorMessage(error));
+  }
+
+  public getSnackErrorMessage(
+    error: HttpErrorResponse | ErrorEvent | string,
+  ): string {
+    if (typeof error === 'string') {
+      return $localize`An error occurred: ${error}`;
+    }
+
+    if (error.error instanceof ErrorEvent) {
+      return $localize`Client error: ${error.error.message}`;
+    }
+
+    if (error instanceof HttpErrorResponse) {
+      // In case of status code 0 or negative, Http module couldn't
+      // connect to the backend
+      if (error.status <= 0) {
+        return $localize`Could not connect to the backend.`;
+      }
+
+      return `[${error.status}] ${this.getBackendErrorLog(error)}\n${
+        error.url
+      }`;
+    }
+
+    if (error instanceof ErrorEvent) {
+      return error.message;
+    }
+
+    return $localize`Unexpected error encountered`;
+  }
+
+  public getBackendErrorLog(error: HttpErrorResponse): string {
+    if (error.error === null) {
+      return error.message;
+    } else {
+      // Show the message the backend has sent
+      return error.error.log ? error.error.log : error.error.error;
+    }
+  }
+
+  public getErrorMessage(
+    error: HttpErrorResponse | ErrorEvent | string,
+  ): string {
+    if (typeof error === 'string') {
+      return error;
+    }
+
+    if (error instanceof HttpErrorResponse) {
+      if (this.getBackendErrorLog(error) !== undefined) {
+        return this.getBackendErrorLog(error);
+      }
+
+      return `${error.status}: ${error.message}`;
+    }
+
+    if (error instanceof ErrorEvent) {
+      return error.message;
+    }
+
+    return `Unexpected error encountered`;
+  }
+}
```

Comment:
@orfeas-k I would prefer to move this code to an interceptor and keep this service cleaner. You will also don't need to remember to add `catchError(error => this.handleError(error))` in every new request added to the `CDBBackendService` service.